pycloudlib/gce/cloud.py

```diff
@@ -50,6 +52,21 @@ def __init__(
 
         if project:
             os.environ["GOOGLE_CLOUD_PROJECT"] = str(project)
+        else:
+            command = ['gcloud', 'config', 'get-value', 'project']
+            exception_text = (
+                "Could not obtain GCE project id. Has the CLI client been "
+                "setup?\nCommand attempted: '{}'".format(' '.join(command))
+            )
+            try:
+                result = subp(command)
+            except FileNotFoundError as e:
+                raise Exception(exception_text) from e
+            if not result.ok:
```

Comment:
Hmm, looking at this again, can we actually take this `if` path? `not result.ok` implies a non-zero return code (of course), but we aren't passing `rcs` to `subp`, so it will default to `(0,)`: https://github.com/canonical/pycloudlib/blob/8a9a79792ead5b680dd558988732b3b533d60b0f/pycloudlib/util.py#L147 which I believe will in turn mean that we get a `RuntimeError` raised any time `result.ok` would be `False`: https://github.com/canonical/pycloudlib/blob/8a9a79792ead5b680dd558988732b3b533d60b0f/pycloudlib/util.py#L193-L201

Reply:
Yeah, that looks right. I tested the FileNotFoundException but couldn't think of a simple way to test the non-0 condition. The way we layer Result with subp isn't the most intuitive.

Reply:
Updated the `rcs` argument so resolving this